crates/engine/tree/src/tree/mod.rs

```diff
@@ -2224,13 +2224,47 @@ where
 
         let exec_time = Instant::now();
 
-        // TODO: create StateRootTask with the receiving end of a channel and
-        // pass the sending end of the channel to the state hook.
-        let noop_state_hook = |_state: &EvmState| {};
+        let persistence_not_in_progress = !self.persistence_state.in_progress();
+
+        // TODO: uncomment to use StateRootTask
+
+        // let (state_root_handle, state_hook) = if persistence_not_in_progress {
+        //     let consistent_view = ConsistentDbView::new_with_latest_tip(self.provider.clone())?;
+        //
+        //     let state_root_config = StateRootConfig::new_from_input(
+        //         consistent_view.clone(),
+        //         self.compute_trie_input(consistent_view, block.header().parent_hash())
+        //             .map_err(ParallelStateRootError::into)?,
+        //     );
+        //
+        //     let provider_ro = consistent_view.provider_ro()?;
+        //     let nodes_sorted = state_root_config.nodes_sorted.clone();
+        //     let state_sorted = state_root_config.state_sorted.clone();
+        //     let prefix_sets = state_root_config.prefix_sets.clone();
+        //     let blinded_provider_factory = ProofBlindedProviderFactory::new(
+        //         InMemoryTrieCursorFactory::new(
+        //             DatabaseTrieCursorFactory::new(provider_ro.tx_ref()),
+        //             &nodes_sorted,
+        //         ),
+        //         HashedPostStateCursorFactory::new(
+        //             DatabaseHashedCursorFactory::new(provider_ro.tx_ref()),
+        //             &state_sorted,
+        //         ),
+        //         prefix_sets,
+        //     );
+        //
+        //     let state_root_task = StateRootTask::new(state_root_config,
+        // blinded_provider_factory);     let state_hook = state_root_task.state_hook();
+        //     (Some(state_root_task.spawn(scope)), Box::new(state_hook) as Box<dyn OnStateHook>)
+        // } else {
+        //     (None, Box::new(|_state: &EvmState| {}) as Box<dyn OnStateHook>)
+        // };
+        let state_hook = |_state: &EvmState| {};
```

Comment:
to match what will come from the now commented code
```suggestion
        let state_hook = Box::new(|_state: &EvmState| {});
```